### DISCUSSION TITLE
关于模板在overleaf上使用问题

### DISCUSSION THREAD

**Discussion by neilwen987** [note286/xduts#76] [Q&A] [answered]
想询问一下如何将模板上传到overleaf使用呢？按照readme教程，下载出来的[the document class/package files](https://mirrors.cloud.tencent.com/CTAN/systems/texlive/tlnet/archive/xduts.tar.xz),应该上传到project的哪里呢？ 按照网上教程，尝试直接点击上传项目
![image](https://user-images.githubusercontent.com/56498265/226114396-3c37cc10-0afe-451f-9663-bd3347642cb7.png)
，但是文件并非.zip结尾，上传失败。
我随后又新建了一个project， 将xduts.tar.xz文件上传在目录下，还是没有办法调用所需要的库。TAT
希望可以完善一下配置overleaf部分的教程

**Comment by note286** [comment, marked as answer]
下载之后并解压，你会获得

```
xdufont.sty
xdulogo.pdf
xdupgthesis.cls
xduugthesis.cls
xduugtp.cls
```

将你需要的上传即可，至于怎么使用，阅读`xduts.pdf`即可。

至于你说的教程，如果解压还需要我教的话，有点过分了啊。

**Reply by neilwen987** [reply]
感谢，readme里面写的直接上传压缩文件，可能我理解的有些歧义

**Reply by note286** [reply]
> Finally, upload the document class/package files to the project on Overleaf/TeXPage.

你阅读的不仔细，`files`而不是`archive`，而且`document class/package files`指的是`.cls`或者`.sty`文件，这是LaTeX术语，即文档类或宏包文件。README中为了精简篇幅，基本没有冗余的话，所以，阅读需要认真点或者说灵活点。

**Reply by neilwen987** [reply]
My bad.主要是第二行相同的词语这里给了一个link, 就让我很容易误解为下面的document class/package files指代了这个压缩文件
![image](https://user-images.githubusercontent.com/56498265/226115251-63284ba2-87a4-4424-846d-9ec73c3ac548.png)

**Reply by note286** [reply]
XDUTS这个项目从README到PDF手册，基本都这样，很多介绍并没有掰开了揉碎了去讲。如果毫无基础的用户，看不懂很正常，我也没指望这些用户能看懂，看不懂也没关系，并不影响使用。XDUTS的设计就是只要你能走到编译那一步，你就能编译出PDF，你就可以开始写你论文的内容了，至于我提供的各种功能，看不懂就不用，没关系。最直接的原因就是我不想把这个项目弄得不伦不类，我希望README和手册里介绍的都是和XDUTS直接相关的内容，而不是参杂着LaTeX入门教程。如果没有基础，应该去看[《一份（不太）简短的LaTeX2ε介绍》](http://mirrors.ctan.org/info/lshort/chinese/lshort-zh-cn.pdf)，这一点手册里是提到的，有指路。就像手册中关于字体的部分，有些字体需要额外安装才可以使用，在手册中我仅仅提及了对应的字体文件名，至于如何安装，这属于LaTeX基础知识，不属于XDUTS专属的内容，所以我就不会在手册中介绍。

至于你本次提问的，我是默认用户需要有一定的LaTeX常识，知道`.cls`、`.sty`和`.tex`的区别，个人觉得吧，我的前提假设并不算过分。不过也有的人会说对新手不够友好，还是那句话，这是XDUTS项目，这个项目是提供一个面向西电学子的有良好接口的LaTeX模板。如果对LaTeX几乎零基础，那么用户应该去学习基础知识，毕竟LaTeX虽然相对小众，但是入门教程还是大把的。就像如果你用一个Java库，总不能指望这个库的手册去教你怎么安装或者Java基础知识吧，除非该库的安装与常规操作不同。总不会指望去一个库的手册里去学该语言的基础知识吧。吐槽几句，我总觉得很多LaTeX项目把用户惯坏了，README里啥都写，不过考虑到LaTeX现状，也算能理解。

> 主要是第二行相同的词语这里给了一个link, 就让我很容易误解为下面的document class/package files指代了这个压缩文件

你指出的这个，可以说是我不严谨，有歧义。直接原因还是我不想啰里啰唆，事后找补的原因可以说是给用户设置一定的门槛。其实你要是不提问，自己折腾几分钟，估计也弄明白了。不是说你不该提问，我还是很欢迎用户与我交流的。只要用户问，无论多么极端，在我这里都会得到回复的，虽然有的我还是会吐槽几句，毕竟我也是个人，不是AI。之前还有用户直接编译`.cls`的，各种你想象不出来的操作都有，还挺有意思的。
